pkg/app/pipedv1/cmd/piped/grpcapi/plugin_api.go

```diff
@@ -143,3 +159,58 @@ func (a *PluginAPI) PutDeploymentPluginMetadataMulti(ctx context.Context, req *s
 func (a *PluginAPI) GetDeploymentSharedMetadata(ctx context.Context, req *service.GetDeploymentSharedMetadataRequest) (*service.GetDeploymentSharedMetadataResponse, error) {
 	return a.metadataStoreRegistry.GetDeploymentSharedMetadata(ctx, req)
 }
+
+func (a *PluginAPI) NotifyWaitApproval(ctx context.Context, req *service.NotifyWaitApprovalRequest) (*service.NotifyWaitApprovalResponse, error) {
+	users, groups, err := getMentionTargets(ctx, model.NotificationEventType_EVENT_DEPLOYMENT_WAIT_APPROVAL, req.Deployment.Id, a.metadataStoreRegistry)
+	if err != nil {
+		return nil, err
+	}
+
+	a.notifier.Notify(model.NotificationEvent{
+		Type: model.NotificationEventType_EVENT_DEPLOYMENT_WAIT_APPROVAL,
+		Metadata: &model.NotificationEventDeploymentWaitApproval{
+			Deployment:        req.Deployment,
+			MentionedAccounts: users,
+			MentionedGroups:   groups,
+		},
+	})
+	return &service.NotifyWaitApprovalResponse{}, nil
+}
+
+func (a *PluginAPI) NotifyApproved(ctx context.Context, req *service.NotifyApprovedRequest) (*service.NotifyApprovedResponse, error) {
+	users, groups, err := getMentionTargets(ctx, model.NotificationEventType_EVENT_DEPLOYMENT_APPROVED, req.Deployment.Id, a.metadataStoreRegistry)
+	if err != nil {
+		return nil, err
+	}
+	a.notifier.Notify(model.NotificationEvent{
+		Type: model.NotificationEventType_EVENT_DEPLOYMENT_APPROVED,
+		Metadata: &model.NotificationEventDeploymentApproved{
+			Deployment:        req.Deployment,
+			Approver:          strings.Join(req.Approvers, ","),
+			MentionedAccounts: users,
+			MentionedGroups:   groups,
+		},
+	})
+	return &service.NotifyApprovedResponse{}, nil
+}
+
+// getMentionTargets returns the list of users and groups who should be mentioned in the notification.
+func getMentionTargets(ctx context.Context, e model.NotificationEventType, deploymentID string, msr *metadatastore.MetadataStoreRegistry) (users []string, groups []string, err error) {
+	n, err := msr.GetDeploymentSharedMetadata(ctx, &service.GetDeploymentSharedMetadataRequest{
+		DeploymentId: deploymentID,
+		Key:          model.MetadataKeyDeploymentNotification,
+	})
+	if err != nil {
+		return []string{}, []string{}, err
+	}
+	if !n.Found {
+		return []string{}, []string{}, nil
+	}
```

Comment:
[nits]
We can use `nil` as the zero-value of slices.
```suggestion
	if err != nil {
		return nil, nil, err
	}
	if !n.Found {
		return nil, nil, nil
	}
```

Reply:
@Warashi Thank you! I fixed it here https://github.com/pipe-cd/pipecd/pull/5613/commits/49402ed3b675b5680b0c8d66db164a19fd882860